src/components/PostDetails.tsx

```diff
@@ -1,106 +1,154 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import * as serverAction from '../api/api';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  post: Post | null;
+};
+
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [isCreateComment, setIsCreateComment] = useState(false);
+  const [isDeleteError, setIsDeleteError] = useState(false);
+
+  const loadComments = () => {
+    if (post && !isError) {
+      setIsCreateComment(false);
+      setIsLoading(true);
+
+      serverAction
+        .getComments(post.id)
+        .then(result => {
+          const results = result.map(comment => ({
+            id: comment.id,
+            postId: comment.postId,
+            name: comment.name,
+            body: comment.body,
+            email: comment.email,
+          }));
+
+          setComments(results);
+        })
+        .catch(() => {
+          setIsError(true);
+          setTimeout(() => {
+            setIsError(false);
+          }, 3000);
```

Comment:
The error handling here is basic and only sets a generic error state. Consider providing more detailed feedback to the user about what went wrong, such as displaying specific error messages based on the error type.